src/platform/common/configuration/service.base.ts

```diff
@@ -8,6 +8,9 @@ import { JupyterSettings } from '../configSettings';
 import { isTestExecution, isUnitTestExecution } from '../constants';
 import { IConfigurationService, IWatchableJupyterSettings } from '../types';
 
+/**
+ * Wrapper around the workspace.getConfiguration api. Makes for typesafe access to configuration properties.
```

Comment:
Are they not typesafe out of the box because the configurations are specific to the extension?

Reply:
Yes. The workspaceConfiguration from VS code allows any type for a value.